### PR TITLE
Upgrade go-jsonnet to to v0.18.0

### PR DESCRIPTION
**Pull request by groodt**

### CODE REVIEW

**Comment by groodt** [comment]
Any thoughts on this CI failure @Globegitter @sparkprime ?

golang toolchain issue or new bazel 5.0 version?

**Comment by UebelAndre** [comment]
https://github.com/bazelbuild/rules_jsonnet/pull/152#issuecomment-1014023152 was what I thought the issue was.

**Comment by groodt** [comment]
> [#152 (comment)](https://github.com/bazelbuild/rules_jsonnet/pull/152#issuecomment-1014023152) was what I thought the issue was.

Ah! Thanks! I didn't see somebody else tried to bump this.

**Comment by UebelAndre** [comment]
> Ah! Thanks! I didn't see somebody else tried to bump this.

hehe all good. I only closed the PR to reduce the number of things I'm waiting on and didn't know if there were any maintainers around. But I do think the issue is in the upstream repositories trying to setup a host installed SDK. If I had bandwidth I'd try to change that and get a patch release out.

**Comment by UebelAndre** [comment]
@groodt would you be able to update [@google_jsonnet_go//bazel:deps.bzl](https://github.com/google/go-jsonnet/blob/v0.18.0/bazel/deps.bzl#L14) to either not use `host` or allow for pass-through arguments from the parent macro?

**Comment by groodt** [comment]
@UebelAndre Yes, I think I should be able to get to that over the next few days. I was planning on pass-through of a version string from `jsonnet_go_dependencies`.

**Comment by groodt** [comment]
This is now 💚  in CI @UebelAndre @sparkprime @Globegitter 

I'll update the references to `go-jsonnet` once this is merged https://github.com/google/go-jsonnet/pull/595

File: jsonnet/jsonnet.bzl [hunk 1]
```diff
@@ -840,16 +840,16 @@ def jsonnet_repositories():
...
         ],
     )
 
     http_archive(
         name = "google_jsonnet_go",
-        sha256 = "4fd04d0c9e38572ef388d28ea6b1ac151b8a9a5026ff94e3a68bdbc18c4db38a",
-        strip_prefix = "go-jsonnet-0.17.0",
-        urls = ["https://github.com/google/go-jsonnet/archive/refs/tags/v0.17.0.tar.gz"],
+        sha256 = "20fdb3599c2325fb11a63860e7580705590faf732abf47ed144203715bd03a70",
+        strip_prefix = "go-jsonnet-0d78479d37eabd9451892dd02be2470145b4d4fa",
```
Comment (UebelAndre) [inline jsonnet/jsonnet.bzl:853]: Small nit, can you add a comment about what version this is closes to?
```suggestion
        # go-jsonnet-0.18.0
        strip_prefix = "go-jsonnet-0d78479d37eabd9451892dd02be2470145b4d4fa",
```

Comment (groodt) [reply jsonnet/jsonnet.bzl:853]: I'd be happy to. Im not sure what the maintainers preference is. It appears in the past that all 3 of (jsonnet, go-jsonnet and rules_jsonnet) were versioned in unison. I personally think that makes sense for jsonnet and go-jsonnet, but not rules_jsonnet. However, I'm not a maintainer.

My suggestion for a comment would be:
`# https://github.com/google/go-jsonnet/commit/0d78479d37eabd9451892dd02be2470145b4d4fa`

I'll wait for the maintainers to provide their input.

Comment (UebelAndre) [reply jsonnet/jsonnet.bzl:853]: It'd be nice if there was a `0.18.1` or something, but I find it nice to be able to recognize at a glance, generally what version I'm using when a commit hash is used as the pin. Again though, small nit, not a big deal. Thanks for working on this!

**Comment by groodt** [comment]
Any chance you can take a look @Globegitter @sparkprime ?

**Pull request merged**